presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -101,7 +104,7 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
 
         TupleDomain<HiveColumnHandle> simplifiedDynamicFilter = dynamicFilter
                 .getCurrentPredicate()
-                .transform(HiveColumnHandle.class::cast).simplify();
+                .transform(HiveColumnHandle.class::cast).simplify(domainCompactionThreshold);
```

Comment:
this requires benchmarking with unpartitioned data set, as discrete values set is extracted in readers (e.g in `TupleDomainOrcPredicate`). The more values in the set, the more CPU cost per stripe.

Reply:
I'm willing to benchmark it. Since this is the same limit that is used for static predicates and the same overheads apply in that case too, I assumed that at some point it was established that this threshold is reasonable.

Reply:
> I'm willing to benchmark it.

Sounds good to me. 

> Since this is the same limit that is used for static predicates and the same overheads apply in that case too

Yes. It's reasonable assumption.